src/Pelagos/Bundle/AppBundle/Form/DatasetSubmissionType.php

```diff
@@ -306,9 +314,30 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             if ($data) {
                 if ($data->isDatasetFileInColdStorage() === true) {
                     $form->get('isDatasetFileInColdStorage')->setData(true);
+                    $form->get('datasetFileColdStorageArchiveSize')->setData(
+                        $data->getDatasetFileColdStorageArchiveSize()
+                    );
+                    $form->get('datasetFileColdStorageArchiveSha256Hash')->setData(
+                        $data->getDatasetFileColdStorageArchiveSha256Hash()
+                    );
+                    $form->get('datasetFileColdStorageOriginalFilename')->setData(
+                        $data->getDatasetFileColdStorageOriginalFilename()
+                    );
                 }
             }
         });
+
+        $builder->addEventListener(
+            FormEvents::POST_SUBMIT,
+            function (FormEvent $event) {
+                $entity = $event->getForm()->getData();
+                $entity->setDatasetFileColdStorageAttributes(
```

Comment:
Dataset Submission tool fails, because of type error thrown by the cold storage setters.